bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/ircc/IrccProtocol.java

```diff
@@ -0,0 +1,822 @@
+            logger.debug("IRCC service returned a 500 - probably an unknown command: {}", cmdToSend);
+        } else {
+            logger.debug("Bad return code from {}: {}", IrccClient.SRV_ACTION_SENDIRCC, resp);
+        }
+    }
+
+    /**
+     * Set's the content URL of the device
+     *
+     * @param contentUrl the non-null, non-empty content url
+     */
+    public void sendContentUrl(final String contentUrl) {
+        Validate.notEmpty(contentUrl, "contentUrl cannot be empty");
+
+        final String sendContentUrl = irccClient.getUrlForAction(IrccClient.AN_SENDCONTENTURL);
+        if (sendContentUrl == null || StringUtils.isEmpty(sendContentUrl)) {
+            logger.debug("{} action was not implmented", IrccClient.AN_SENDCONTENTURL);
+            return;
+        }
+        final String body = "<contentUrl><url>" + contentUrl + "</url></contentUrl>";
+        final HttpResponse resp = transport.executePostXml(sendContentUrl, body);
+        if (resp.getHttpCode() == HttpStatus.OK_200) {
+            logger.trace("Send of URL {} was successful", contentUrl);
+            // Do nothing
+        } else if (resp.getHttpCode() == HttpStatus.SERVICE_UNAVAILABLE_503) {
+            logger.debug("IRCC service is unavailable (power off?)");
+        } else {
+            logger.debug("Bad return code from {}: {}", IrccClient.AN_SENDCONTENTURL, resp);
+        }
+    }
+
+    /**
+     * Set's the text field with the specified text
+     *
+     * @param text the non-null, non-empty text
+     */
+    public void sendText(final String text) {
+        Validate.notEmpty(text, "text cannot be empty");
+        final String sendTextUrl = irccClient.getUrlForAction(IrccClient.AN_SENDTEXT);
+        if (StringUtils.isEmpty(sendTextUrl)) {
+            logger.debug("{} action was not implmented", IrccClient.AN_SENDTEXT);
+            return;
+        }
+
+        try {
+            final String textParm = "?text=" + URLEncoder.encode(text, "UTF-8");
+            final HttpResponse resp = transport.executeGet(sendTextUrl + textParm);
+            if (resp.getHttpCode() == HttpStatus.OK_200) {
+                logger.trace("Send of text {} was successful", text);
+            } else if (resp.getHttpCode() == HttpStatus.NOT_ACCEPTABLE_406) {
+                logger.debug("{} was sent but 'not acceptable' was returned (ie no input field to accept text)",
+                        IrccClient.AN_SENDTEXT);
+            } else if (resp.getHttpCode() == HttpStatus.SERVICE_UNAVAILABLE_503) {
+                logger.debug("IRCC service is unavailable (power off?)");
+            } else {
+                logger.debug("Unknown code for {}:L {}", IrccClient.AN_SENDTEXT, resp);
+            }
+        } catch (final UnsupportedEncodingException e) {
+            logger.debug("UTF-8 is not supported on this platform: {}", e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void close() {
+    }
```

Comment:
don't you need to close `transport` here?

Reply:
Good catch